arangod/Pregel/GraphStore/GraphSerdeConfigBuilderSingleServer.cpp

```diff
@@ -0,0 +1,97 @@
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Pregel/GraphStore/GraphSerdeConfigBuilderSingleServer.h"
+
+#include "Containers/Enumerate.h"
+#include "VocBase/LogicalCollection.h"
+
+namespace arangodb::pregel {
+
+GraphSerdeConfigBuilderSingleServer::GraphSerdeConfigBuilderSingleServer(
+    TRI_vocbase_t& vocbase, GraphByCollections const& graphByCollections)
+    : vocbase(vocbase), graphByCollections(graphByCollections) {}
+
+[[nodiscard]] auto
+GraphSerdeConfigBuilderSingleServer::edgeCollectionRestrictionsByShard() const
+    -> ShardMap {
+  return graphByCollections.edgeCollectionRestrictions;
+}
+
+[[nodiscard]] auto GraphSerdeConfigBuilderSingleServer::checkVertexCollections()
+    const -> Result {
+  for (std::string const& name : graphByCollections.vertexCollections) {
+    auto coll = vocbase.lookupCollection(name);
+
+    if (coll == nullptr || coll->deleted()) {
+      return Result{TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, name};
+    }
+  }
+  return Result{};
+}
+
+[[nodiscard]] auto GraphSerdeConfigBuilderSingleServer::checkEdgeCollections()
+    const -> Result {
+  for (std::string const& name : graphByCollections.edgeCollections) {
+    auto coll = vocbase.lookupCollection(name);
+
+    if (coll == nullptr || coll->deleted()) {
+      return Result{TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, name};
+    }
+  }
+  return Result{};
+}
+
+[[nodiscard]] auto GraphSerdeConfigBuilderSingleServer::loadableVertexShards()
+    const -> LoadableVertexShards {
+  auto result = LoadableVertexShards{};
+
+  for (auto const [idx, vertexCollection] :
+       enumerate(graphByCollections.vertexCollections)) {
+    auto loadableVertexShard =
+        LoadableVertexShard{.pregelShard = PregelShard(idx),
+                            .vertexShard = vertexCollection,
+                            .collectionName = vertexCollection,
+                            .edgeShards = {}};
+    for (auto const& edgeCollection : graphByCollections.edgeCollections) {
+      if (not graphByCollections.isRestricted(vertexCollection,
+                                              edgeCollection)) {
+        loadableVertexShard.edgeShards.emplace_back(edgeCollection);
+      }
+    }
+    result.loadableVertexShards.emplace_back(loadableVertexShard);
+  }
+  return result;
+}
+
+[[nodiscard]] auto GraphSerdeConfigBuilderSingleServer::responsibleServerMap(
+    LoadableVertexShards const& loadableVertexShards) const
+    -> ResponsibleServerMap {
+  auto result = ResponsibleServerMap{};
+
+  result.responsibleServerMap.resize(
+      loadableVertexShards.loadableVertexShards.size(), "");
```

Comment:
Is it correct that you don't put anything into result here?

Reply:
but I do, every shard has `""` (the local server) as it's responsible server!